plugins/flytekit-vaex/flytekitplugins/vaex/sd_transformers.py

```diff
@@ -0,0 +1,73 @@
+import os
+import typing
+
+import pandas as pd
+import vaex
+
+from flytekit import FlyteContext, StructuredDatasetType
+from flytekit.models import literals
+from flytekit.models.literals import StructuredDatasetMetadata
+from flytekit.types.structured.structured_dataset import (
+    PARQUET,
+    StructuredDataset,
+    StructuredDatasetDecoder,
+    StructuredDatasetEncoder,
+    StructuredDatasetTransformerEngine,
+)
+
+
+class VaexDataFrameToParquetEncodingHandler(StructuredDatasetEncoder):
+    def __init__(self):
+        super().__init__(vaex.dataframe.DataFrameLocal, None, PARQUET)
+
+    def encode(
+        self,
+        ctx: FlyteContext,
+        structured_dataset: StructuredDataset,
+        structured_dataset_type: StructuredDatasetType,
+    ) -> literals.StructuredDataset:
+        df = typing.cast(vaex.dataframe.DataFrameLocal, structured_dataset.dataframe)
+        path = ctx.file_access.get_random_remote_directory()
+        local_dir = ctx.file_access.get_random_local_directory()
+        local_path = os.path.join(local_dir, f"{0:05}")
+        df.export_parquet(local_path)
+        ctx.file_access.upload_directory(local_dir, path)
+        return literals.StructuredDataset(
+            uri=path,
+            metadata=StructuredDatasetMetadata(structured_dataset_type=StructuredDatasetType(structured_dataset_type)),
+        )
+
+
+class ParquetToVaxDataFrameDecodingHandler(StructuredDatasetDecoder):
+    def __init__(self):
+        super().__init__(vaex.dataframe.DataFrameLocal, None, PARQUET)
+
+    def decode(
+        self,
+        ctx: FlyteContext,
+        flyte_value: literals.StructuredDataset,
+        current_task_metadata: StructuredDatasetMetadata,
+    ) -> vaex.dataframe.DataFrameLocal:
+        local_dir = ctx.file_access.get_random_local_directory()
+        ctx.file_access.get_data(flyte_value.uri, local_dir, is_multipart=True)
+        path = f"{local_dir}/00000"
```

Comment:
I'm wondering if it's okay to consider the first partition if the dataframe is huge. How about we consider all files that are present under the parquet directory using `vaex.open` or `vaex.open_many`? I think you can use the glob pattern.

Reply:
@samhita-alla  Actually looking at this more closely - i think i may need to use `df.export_many` https://vaex.io/docs/guides/io.html#Export-to-multiple-files-in-parallel if we want to serialise chunks  to multiple parts in parallel.

What i implemented written in chunks serially to single parquet according to the docs https://vaex.io/docs/guides/io.html#Exporting-binary-file-formats
"""
When exporting to Apache Arrow and Apache Parquet file format, the data is written in chunks thus enabling to export of data that does not fit in RAM all at once. A custom chunk size can be specified via the chunk_size argument, the default value of which is 1048576. For example:
"""

Do we want to support one or both options and do we want to give the user option to override chunk size ?

Reply:
Also, ive been trying to mimic polars implementation - any idea why `00000` suffix in path and is this being split to multiple parts by flyte (as polars implementation seems to write to single parquet https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.DataFrame.write_parquet.html) ?

https://github.com/flyteorg/flytekit/blob/caf612d37e3767d6d0184d6d4e86c4a01f8adeab/plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py#L66-L67

Reply:
@ryankarlos, thanks for doing the research! I think `00000` isn't a partition after all. It's right in your code where you're creating a local path using `local_path = os.path.join(local_dir, f"{0:05}")` syntax. So it's just a file. 😅

Would you mind creating an issue to support writing large dataframes using `export_many`? This isn't required now but you or someone else can implement it later.